js/util/ajax.js

```diff
@@ -59,24 +59,19 @@ exports.getImage = function(url, callback) {
     return exports.getArrayBuffer(url, (err, imgData) => {
         if (err) return callback(err);
         const img = new window.Image();
-        img.onload = function() {
-            callback(null, img);
-            (window.URL || window.webkitURL).revokeObjectURL(img.src);
-        };
-        const blob = new window.Blob([new Uint8Array(imgData)], { type: 'image/png' });
-        if (imgData.byteLength) {
-            img.src = (window.URL || window.webkitURL).createObjectURL(blob);
-        } else {
-            img.src = transparentPngUrl;
-        }
-        img.getData = function() {
+        img.onload = () => {
             const canvas = window.document.createElement('canvas');
             const context = canvas.getContext('2d');
             canvas.width = img.width;
             canvas.height = img.height;
             context.drawImage(img, 0, 0);
-            return context.getImageData(0, 0, img.width, img.height).data;
+            img.data = context.getImageData(0, 0, img.width, img.height).data;
```

Comment:
Not does it have `getData()` method it was augmented with earlier. There are some places in the code that expect `getImage` to return the image object rather than underlying data, such as `ImageSource`. On the other hand, moving canvas-based getData code to `ImageSprite` will make stubbing in Node more difficult because you have to stub its methods rather than `ajax.getImage`. So I suggest leaving this as is for now.